### PR TITLE
improve deletion  confirmation messages

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: HOMER Etienne <etiennehomer@gmail.com>

### CODE REVIEW

File: src/components/tree-views-container.js [hunk 1]
```diff
@@ -529,6 +529,12 @@ const TreeViewsContainer = () => {
         );
     };
 
+    const getTargetDirectory = () => {
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: _getActiveDirectoryName_ ?

File: src/translations/en.json [hunk 2]
```diff
@@ -51,8 +52,12 @@
...
     "insertNewDirectoryDialogTitle": "Create folder",
     "insertNewRootDirectoryDialogTitle": "Create root folder",
-    "deleteDirectoryDialogMessage": "Are you sure you want to delete this folder?",
-    "deleteDirectoryDialogTitle": "Delete the folder",
+    "deleteDirectoryDialogMessage": "Are you sure you want to delete the {directoryName} folder and the items it contains?",
+    "deleteItemDialogMessage": "Are you sure you want to delete {elementName} ?",
+    "deleteMultipleItemsDialogMessage": "Are you sure you want to delete those {itemsCount} items ?",
+    "deleteDirectoryDialogTitle": "Delete {directoryName} folder",
+    "deleteItemDialogTitle": "Delete item {elementName}",
+    "deleteMultipleItemsDialogTitle": "Delete {itemsCount} items",
```
Comment (sBouzols) [inline src/translations/en.json:61]: Should see plural management in internationalization to limit translations entries
see deleteElementsFailure for exemple

File: src/translations/fr.json [hunk 3]
```diff
@@ -50,8 +51,12 @@
...
     "insertNewDirectoryDialogTitle": "Créer un dossier",
     "insertNewRootDirectoryDialogTitle": "Créer un dossier racine",
-    "deleteDirectoryDialogMessage": "êtes vous sûr de vouloir supprimer le dossier ?",
-    "deleteDirectoryDialogTitle": "Supprimer le dossier",
+    "deleteDirectoryDialogMessage": "Êtes-vous sûr(e) de vouloir supprimer le dossier {directoryName} ainsi que les éléments qu'il contient ?",
+    "deleteItemDialogMessage": "Êtes-vous sûr(e) de vouloir supprimer {elementName} ?",
+    "deleteMultipleItemsDialogMessage": "Êtes-vous sûr(e) de vouloir supprimer ces {itemsCount} éléments ?",
+    "deleteDirectoryDialogTitle": "Supprimer le dossier {directoryName}",
+    "deleteItemDialogTitle": "Supprimer l'élément {elementName}",
+    "deleteMultipleItemsDialogTitle": "Supprimer {itemsCount} éléments",
```
Comment (sBouzols) [inline src/translations/fr.json:60]: same in french

File: src/components/tree-views-container.js [hunk 4]
```diff
@@ -728,15 +730,33 @@ const TreeViewsContainer = () => {
...
+                        )}
+                    </span>
+                }
                 open={openDeleteDirectoryDialog}
                 onClick={deleteSelectedDirectory}
                 onClose={handleCloseDeleteDirectoryDialog}
-                title={intl.formatMessage({
-                    id: 'deleteDirectoryDialogTitle',
-                })}
+                title={intl.formatMessage(
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: must be managed inside the deleteDialog component not in tree-views-container

File: src/components/tree-views-container.js [hunk 5]
```diff
@@ -728,15 +730,33 @@ const TreeViewsContainer = () => {
                 error={renameError}
             />
             <DeleteDialog
-                message={intl.formatMessage({
-                    id: 'deleteDirectoryDialogMessage',
-                })}
+                message={
```
Comment (sBouzols) [inline src/components/tree-views-container.js]: must be managed inside the deleteDialog component not in tree-views-container

File: src/components/directory-content.js [hunk 6]
```diff
@@ -882,13 +937,21 @@ const DirectoryContent = () => {
...
                 onClose={handleCloseDeleteElement}
                 onClick={handleClickDeleteElement}
-                title={useIntl().formatMessage(
-                    { id: 'deleteElement' },
-                    { stn: getSelectedChildren().length }
-                )}
-                message={useIntl().formatMessage({
-                    id: 'genericConfirmQuestion',
-                })}
+                title={
```
Comment (sBouzols) [inline src/components/directory-content.js]: must be managed inside the deleteDialog component not in directory-content

File: src/components/directory-content.js [hunk 7]
```diff
@@ -652,6 +653,60 @@ const DirectoryContent = () => {
         return undefined;
     };
 
+    function buildFilesToDeleteGrid(files) {
```
Comment (sBouzols) [inline src/components/directory-content.js]: must be managed inside the deleteDialog component not in directory-content

File: src/components/dialogs/delete-dialog.js [hunk 8]
```diff
@@ -21,7 +21,7 @@ import React from 'react';
  * @param {EventListener} onClose Event to close the dialog
  * @param {EventListener} onClick Event to submit the deletion
  * @param {String} title Title of the dialog
- * @param {String} message Message of the dialog
+ * @param {Object} message Message of the dialog
```
Comment (sBouzols) [inline src/components/dialogs/delete-dialog.js]: not ok. The deletion messages will mainly have context in this dialog not elsewhere.
I think we should pass an Array of elementsToDelete and then we must build messages here in this component from this prop.

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/tree-views-container.js (hunk 1 above)
Comment (etiennehomer) [reply src/components/tree-views-container.js]: changed

File: src/components/tree-views-container.js (hunk 4 above)
Comment (etiennehomer) [reply src/components/tree-views-container.js]: done

File: src/components/tree-views-container.js (hunk 5 above)
Comment (etiennehomer) [reply src/components/tree-views-container.js]: done

File: src/components/directory-content.js (hunk 6 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: done

File: src/components/directory-content.js (hunk 7 above)
Comment (etiennehomer) [reply src/components/directory-content.js]: done

File: src/components/dialogs/delete-dialog.js (hunk 8 above)
Comment (etiennehomer) [reply src/components/dialogs/delete-dialog.js]: done

**Review by sBouzols — APPROVED** [review]
Ok for me.
Plural internationalization rules could be done in another PR.

File: src/components/dialogs/delete-dialog.js [hunk 9]
```diff
@@ -11,20 +11,30 @@ import DialogContentText from '@material-ui/core/DialogContentText';
...
  * @param {String} error Error message
  */
-const DeleteDialog = ({ open, onClose, onClick, title, message, error }) => {
+const DeleteDialog = ({
+    open,
+    onClose,
+    onClick,
+    items,
+    multipleDeleteFormatMessageId,
+    simpleDeleteFormatMessageId,
```
Comment (AbdelHedhili) [inline src/components/dialogs/delete-dialog.js:35]: missing doc to describe those two

**Pull request merged**